page.go

```diff
@@ -133,6 +133,17 @@ type pgids []pgid
 func (s pgids) Len() int           { return len(s) }
 func (s pgids) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 func (s pgids) Less(i, j int) bool { return s[i] < s[j] }
+func (s pgids) isSorted() bool {
```

Comment:
How much slower? Slower enough that it is worth adding more code to work around it? Numbers would help a lot here.